File: app/src/main/java/com/example/milmi/aplikasikotlin/Adapter.java
package com.example.milmi.aplikasikotlin;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.List;

public class Adapter extends RecyclerView.Adapter<Adapter.AppViewHolder> {

    private Context context;
    private List<Data> listData;

    public Adapter(Context context, List<Data> listData) {
        this.context = context;
        this.listData = listData;
    }


    @NonNull
    @Override
    public AppViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(context);
        View view = inflater.inflate(R.layout.activity_cardview, null);
        return new AppViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull AppViewHolder holder, int position) {
        Data data = listData.get(position);
        holder.judul.setText(data.getJudul());
        holder.genre.setText(data.getGenre());
        holder.rating.setText(data.getRating());
        holder.image.setImageDrawable(context.getResources().getDrawable(data.getGambar()));
    }

    @Override
    public int getItemCount() {
        return listData.size();
    }

    class AppViewHolder extends RecyclerView.ViewHolder{
        ImageView image;
        TextView judul, genre, rating;

        public AppViewHolder(@NonNull View itemView) {
            super(itemView);

            image = itemView.findViewById(R.id.image);
            judul = itemView.findViewById(R.id.judul);
            genre = itemView.findViewById(R.id.genre);
            rating = itemView.findViewById(R.id.rating);


        }
    }
}

File: app/src/main/java/com/example/milmi/aplikasikotlin/Adapter.kt
package com.example.milmi.aplikasikotlin

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.Glide
import kotlinx.android.synthetic.main.activity_cardview.view.*

open class Adapter(val context: Context, val listData: MutableList<Data> = arrayListOf()) : RecyclerView.Adapter<Adapter.AppViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AppViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.activity_cardview, parent,false)
        return AppViewHolder(view)
    }

    override fun onBindViewHolder(holder: AppViewHolder, position: Int) {
        Glide.with(context)
                .asBitmap()
                .load(listData.get(position).gambar)
                .into(holder.image)
        holder.judul.text = listData.get(position).judul
        holder.genre.text = listData.get(position).genre
        holder.rating.text = listData.get(position).rating
        holder.durasi.text = listData.get(position).durasi


    }

    override fun getItemCount(): Int {
        return listData.size
    }

    class AppViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView){
        val image =itemView.findViewById<ImageView>(R.id.image)
        val judul = itemView.findViewById<TextView>(R.id.judul)
        val genre = itemView.findViewById<TextView>(R.id.genre)
        val durasi = itemView.findViewById<TextView>(R.id.durasi)
        val rating = itemView.findViewById<TextView>(R.id.rating)
    }
}

